R/http.R

```diff
@@ -23,9 +24,16 @@ iamHTTP <- function(query, verb = "GET", body = "",
                                         `x-amz-date` = d_timestamp),
                request_body = "",
                key = key, secret = secret)
-        H <- add_headers(`x-amz-date` = d_timestamp, 
-                         `x-amz-content-sha256` = S$BodyHash,
-                         Authorization = S$SignatureHeader)
+        if(token != "") {
+             H <- add_headers(`x-amz-date` = d_timestamp,
```

Comment:
Is there a better way you feel this duplication could be avoiding? Looking at httr's header code, it doesn't appear obvious to me how a conditional like this could be dealt with...